app/dev/controllers/capabilitiesController.js

```diff
@@ -14,19 +18,23 @@ app.controller('CapabilitiesController', ['$scope', '$http', '$rootScope', funct
           url: newCapability.capabilityUrl,
           oauth: newCapability.capabilityOauth
         }
-      }).success(function(response) { $scope.capsInCell.push(response) })
-      .error(function() { console.log("error adding capability") });
+      }).success(function(response) {
+        $scope.capsInCell.push(response)
+      }).error(function() {
+        console.log("error adding capability")
+      });
     }
 
-    $scope.updateCapability = function(capability) {
+    $scope.updateCapability = function(capability, id) {
       $http({
         method: 'POST',
         url: '/capabilities/update/' + capability.id,
         data: { name: capability.name, code: capability.code, url: capability.url, oauth: capability.oauth }
-      }).success(function() {
-        
-      })
-      .error(function() { console.log("error adding capability") });
+      }).success(function(response) {
+        $scope.capsInCell.splice($scope.capsInCell.indexOf(capability), 1, response);
```

Comment:
Passing in 'response' to the success callback, then getting indexOf 'capability' ?

Also, I noticed some weird behavior when we try to get the the indexOf the capability after it comes back from the server.  I found it works better to store $scope.capsInCell.indexOf(capability) in a variable before the AJAX call, then use that variable in the splice().